File: get_feature12.dart
// ignore_for_file: avoid_print

import 'package:sqflite_common_ffi/sqflite_ffi.dart';

void main() async {
  sqfliteFfiInit();
  databaseFactory = databaseFactoryFfi;

  final db = await databaseFactory.openDatabase('features.db');

  final List<Map> features = await db.query(
    'features',
    where: 'id = ?',
    whereArgs: [12],
  );

  if (features.isNotEmpty) {
    final feature = features.first;
    print('ID: ${feature['id']}');
    print('Category: ${feature['category']}');
    print('Name: ${feature['name']}');
    print('Description: ${feature['description']}');
    print('Steps: ${feature['steps']}');
    print('Passes: ${feature['passes'] == 1}');
    print('In Progress: ${feature['in_progress'] == 1}');
    print('Dependencies: ${feature['dependencies']}');
    print('Priority: ${feature['priority']}');
  } else {
    print('Feature #12 not found');
  }

  await db.close();
}

File: get_feature12.js
const sqlite3 = require('sqlite3');
const path = require('path');

const dbPath = path.join(__dirname, 'features.db');
const db = new sqlite3.Database(dbPath);

db.get(
  `SELECT id, category, name, description, steps, passes, in_progress, dependencies, priority
   FROM features WHERE id = 12`,
  (err, row) => {
    if (err) {
      console.log(JSON.stringify({error: err.message}, null, 2));
    } else if (row) {
      const result = {
        id: row.id,
        category: row.category,
        name: row.name,
        description: row.description,
        steps: row.steps ? JSON.parse(row.steps) : [],
        passes: row.passes === 1,
        in_progress: row.in_progress === 1,
        dependencies: row.dependencies ? JSON.parse(row.dependencies) : [],
        priority: row.priority
      };
      console.log(JSON.stringify(result, null, 2));
    } else {
      console.log(JSON.stringify({error: "Feature #12 not found"}, null, 2));
    }
    db.close();
  }
);
